src/components/weapons/lock_on_weapon.rs

```diff
@@ -0,0 +1,10 @@
+use super::ranged_weapon::RangedWeapon;
+use crate::components::starships::starship::Starship;
+use bevy::{ecs::component::Component, transform::components::Transform};
+
+// TODO idea for targetting system
+#[derive(Component, Debug, PartialEq)]
+pub struct LockOnWeapon {
```

Comment:
What do you think of this idea for a possible path to go down for weapon targetting. On starship there is a method to check if it "is_destroyed" which you can use to remove the targetting square. The translation in transform can be used to get the current position of the starship.